### PR TITLE
vinyl: add the test cases for the disk layout

### PR DESCRIPTION
**Pull request by rtsisyk**
Hi,

I patched Lua xlog reader to support Vinyl's files and added the test cases.

#2100

### CODE REVIEW

File: src/box/lua/xlog.c [hunk 1]
```diff
@@ -75,20 +75,34 @@ lbox_checkcursor(struct lua_State *L, int narg, const char *src)
...
+static void
+lbox_xlog_pushkey(lua_State *L, const char *key)
+{
+	luaL_Buffer b;
+	luaL_buffinit(L, &b);
+	for (const char *pos = key; *pos; pos++)
+		luaL_addchar(&b, (*pos != ' ') ? *pos : '_');
+	luaL_pushresult(&b);
+}
+
```
Comment (locker) [inline src/box/lua/xlog.c]: Why not just replace all whitespaces with underscores in iproto_key_strs?

File: src/box/lua/xlog.c [hunk 2]
```diff
@@ -188,10 +188,10 @@ lbox_xlog_parser_iterate(struct lua_State *L)
 	assert(rc == 0);
 
 	lua_pushinteger(L, row.lsn);
-	lua_newtable(L);
+	lua_createtable(L, 0, 8);
 	lua_pushstring(L, "HEADER");
 
-	lua_newtable(L);
+	lua_createtable(L, 0, 8);
```
Comment (locker) [inline src/box/lua/xlog.c]: Looks like black magic to me. And what's the point of this change anyway?

File: test/vinyl/layout.test.lua [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        end
+        table.insert(result, { path, rows })
+    end
+end;
+
+test_run:cmd("setopt delimiter ''");
+test_run:cmd("push filter 'timestamp: .*' to 'timestamp: <timestamp>'")
+result
+test_run:cmd("clear filter")
+box.snapshot() -- discard old checkpoints
```
Comment (locker) [inline test/vinyl/layout.test.lua]: What for?

File: src/box/lua/xlog.c (hunk 1 above)
Comment (rtsisyk) [inline src/box/lua/xlog.c]: I asked the same question in Tarantool Dev and @kostja said what whitespaces are better for error messages. As result, I decided to keep whitespaces in iproto_key_strs for error messages and use underscores in Lua.

File: src/box/lua/xlog.c [hunk 4]
```diff
@@ -95,7 +95,7 @@ lbox_xlog_parse_body_kv(struct lua_State *L, int type, const char **beg, const c
 	if (mp_typeof(**beg) != MP_UINT)
 		luaL_error(L, "Broken type of body key");
 	uint32_t v = mp_decode_uint(beg);
-	if (iproto_key_name(v)) {
+	if (iproto_type_is_dml(type) && iproto_key_name(v)) {
```
Comment (rtsisyk) [inline src/box/lua/xlog.c]: `VY_INDEX_RUN_INFO`, `VY_INDEX_PAGE_INFO` and `VY_RUN_PAGE_INDEX` xrow types have their own namespaces for xrow->body keys. For example, there is `IPROTO_SYNC = 0x01`, but at the same time we have `VY_PAGE_INFO_OFFSET = 1`, `VY_RUN_INFO_MIN_LSN = 1` and `VY_PAGE_INDEX_INDEX = 1`. This change checks that request is DML (e.g REPLACE/INSERT/DELETE/UPSERT) first and only after that tries to use iproto_key_name().

File: src/box/lua/xlog.c [hunk 5]
```diff
@@ -272,7 +272,7 @@ lbox_xlog_parser_open_pairs(struct lua_State *L)
 	    strncmp(cur->meta.filetype, "RUN", 3) != 0 &&
 	    strncmp(cur->meta.filetype, "INDEX", 5) != 0 &&
 	    strncmp(cur->meta.filetype, "DATA", 4) != 0 &&
```
Comment (rtsisyk) [inline src/box/lua/xlog.c]: We had `.vyrun`/`.vyindex` vs `.run`/`.index` discussion some time ago and decided to go with `.run`/`.index` and use the same meta.filetype as the file extension.

File: test/vinyl/layout.test.lua [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+space:replace({4})
+space:replace({5})
+space:replace({6})
+box.snapshot()
+space:drop()
+
+-- Get the list of files from the last checkpoint.
+-- convert names to relative
+work_dir = require('fio').cwd()
+files = box.backup.start()
```
Comment (rtsisyk) [inline test/vinyl/layout.test.lua]: OOPS, I'll fix that.

File: test/vinyl/layout.test.lua (hunk 3 above)
Comment (rtsisyk) [inline test/vinyl/layout.test.lua]: Actually this is not needed, I'll remove it.

File: src/box/lua/xlog.c (hunk 5 above)
Comment (locker) [inline src/box/lua/xlog.c]: And what about DATA? I don't see it is used anywhere. May be, just drop it?

**Pull request closed without merging**